sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -148,10 +148,10 @@ package object util extends Logging {
   }
 
   def quoteIfNeeded(part: String): String = {
```

Comment:
This is a pretty common method. Could you double check if it's okay to change this behavior for other places?

Reply:
I think it's only for error message or EXPLAIN result (e.g. `Attribute.sql`). @sarutak can you confirm?

Reply:
I'm now confirming but, for now,  I don't think this change breaks existing behavior except for display format.

Reply:
I looked into the following classes which use `quoteIfNeeded`

* ResolveSessionCatalog.apply
* IdentifierHelper.quoted
* MultipartIdentifierHelper.quoted
* DatabaseInSessionCatalog$.unapply
* NamespaceHelper.quoted
* Alias.sql
* AttributeReference.sql
* UnresolvedAttribute.sql
* IdentifierImpl.toString
* IdentifierHelper.quoted

Finally, I think this change doesn't break existing behavior.